doc/pbh/pbh-design.md

```diff
@@ -0,0 +1,941 @@
+    {
+        attr.id = SAI_ACL_TABLE_ATTR_ACL_STAGE;
+        attr.value.s32 = SAI_ACL_STAGE_INGRESS;
+        table_attrs.push_back(attr);
+
+        attr.id = SAI_ACL_TABLE_ATTR_FIELD_GRE_KEY;
+        attr.value.booldata = true;
+        table_attrs.push_back(attr);
+
+        attr.id = SAI_ACL_TABLE_ATTR_FIELD_ETHER_TYPE;
+        attr.value.booldata = true;
+        table_attrs.push_back(attr);
+
+        attr.id = SAI_ACL_TABLE_ATTR_FIELD_IP_PROTOCOL;
+        attr.value.booldata = true;
+        table_attrs.push_back(attr);
+
+        attr.id = SAI_ACL_TABLE_ATTR_FIELD_IPV6_NEXT_HEADER;
+        attr.value.booldata = true;
+        table_attrs.push_back(attr);
+
+        attr.id = SAI_ACL_TABLE_ATTR_FIELD_L4_DST_PORT;
+        attr.value.booldata = true;
+        table_attrs.push_back(attr);
+
+        attr.id = SAI_ACL_TABLE_ATTR_FIELD_INNER_ETHER_TYPE;
+        attr.value.booldata = true;
+        table_attrs.push_back(attr);
+
+        sai_status_t status = sai_acl_api->create_acl_table(&m_oid, gSwitchId, (uint32_t)table_attrs.size(), table_attrs.data());
+
+        if (status == SAI_STATUS_SUCCESS)
+        {
+            gCrmOrch->incCrmAclUsedCounter(CrmResourceType::CRM_ACL_TABLE, SAI_ACL_STAGE_INGRESS, SAI_ACL_BIND_POINT_TYPE_PORT);
+            gCrmOrch->incCrmAclUsedCounter(CrmResourceType::CRM_ACL_TABLE, SAI_ACL_STAGE_INGRESS, SAI_ACL_BIND_POINT_TYPE_LAG);
+        }
+
+        return status == SAI_STATUS_SUCCESS;
+    }
+
+    ...
+}
+
+class AclRulePbh: public AclRule
+{
+public:
+    AclRulePbh(AclOrch *pAclOrch, string rule, string table, bool createCounter = false);
+
+    bool validateAddPriority(const sai_uint32_t &value);
+    bool validateAddMatch(const sai_attribute_t &attr);
+    bool validateAddAction(const sai_attribute_t &attr);
+    bool validate() override;
+    void update(SubjectType, void *) override;
+};
+```
+
+## 2.4 DB schema
+
+### 2.4.1 Config DB
+
+#### 2.4.1.1 PBH table
+```abnf
+; defines schema for PBH table configuration attributes
+key = PBH_TABLE|table_name ; table name. Must be unique
+
+; field        = value
+interface_list = interface-list ; interfaces to which this table is applied
+description    = 1*255VCHAR     ; table description
+
+; value annotations
+port-name      = 1*64VCHAR                                   ; name of the port
+lag-name       = "PortChannel" 1*4DIGIT                      ; name of the portchannel
+interface-name = port-name / lag-name                        ; name of the interface
+interface-list = interface-name [ 1*( "," interface-name ) ] ; list of the interfaces. Valid values range is platform dependent
+```
+
+**Note:** at least one member of _interface_list_ is required
+
+#### 2.4.1.2 PBH rule
+```abnf
+; defines schema for PBH rule configuration attributes
+key = PBH_RULE|table_name|rule_name ; rule name. Must be unique across the table
+
+; field           = value
+priority         = 1*5DIGIT      ; rule priority. Valid values range is platform dependent
```

Comment:
Could we list order of evaluation of priorities, eg: higher value priority will evaluate before lower value priority in hw?

Reply:
@anish-n done